### ISSUE TITLE
Fix bugs related to the scripture pane  highlighted words generation

### ISSUE THREAD

**Issue by mannycolon** [unfoldingWord/translationCore#5185]
### User Story

As a user, I want to make a selection in tW without it crashing so that I can get work done.

### Features / Specifications

- Fix tW crashing when a selection is made.
- [ ]
- [ ]

### Definition of Done

- tW does not crash.

**Comment by benjore** [comment]
@mannycolon Please add description.

**Comment by da1nerd** [comment]
note: though linked to a PR with some fixes, there's still a bug in the tc-ui-toolkit. 

The exact cause of the issue is unclear to me. However, part of it appears to originate in `src/ScripturePane/helpers`  in the `getCurrentVersesWithHighlightedWords` function.

The result of this bug is the scripture pane fails to load in either tool and the groups menu is missing most of the navigation.

**Comment by cckozie** [comment]
I don't know how to test this. What steps need to be followed to observe the problem?

**Comment by da1nerd** [comment]
@cckozie the issue ended up mostly being not all of the code being merged in or tc using the wrong version of the tools.

To test one specific bug you should open tW in the latest `1.0.2` build and make a text selection. Previously the application was crashing due to a bug introduced **after** `1.0`. Now the application should continue to function without crashing.

That should be all you need to test for this issue.

**Comment by cckozie** [comment]
1.0.2 (6ab77c0) 
Per Joel's instructions above, I loaded a couple of projects in tW to verified that I could make selections and save them. Some worked and some did not, although I did not see any crashes.
1. Import this project: 
[53-1TH.usfm.zip](https://github.com/unfoldingWord-dev/translationCore/files/2470277/53-1TH.usfm.zip)
2. Load it with tW
3. In the first check (amen, verse 3:3) double click on 'Amen' to select it.
4. Click Save
5. Click Save & Continue
6. Navigate to the group for 'Christ, Messiah'
7. Double click on 'Christ' to select it.
8. Click Save
Notice that the 'Some selections are not longer valid...' alert is displayed, the broken link icon is displayed next to the verse in the menu, and that the selection is not saved.

I don't know if it's related or not, but I am seeing this in the console:
![image](https://user-images.githubusercontent.com/8171759/46827447-35eacb80-cd5e-11e8-991f-647c2d205e8e.png)
